qiskit_nature/operators/second_quantization/vibrational_op.py

```diff
@@ -0,0 +1,338 @@
+
+        label_list, indices = np.unique(self._labels, return_inverse=True, axis=0)
+        coeff_list = np.zeros(len(self._coeffs), dtype=np.complex128)
+        for i, val in zip(indices, self._coeffs):
+            coeff_list[i] += val
+        non_zero = [
+            i for i, v in enumerate(coeff_list) if not np.isclose(v, 0, atol=atol, rtol=rtol)
+        ]
+        if not non_zero:
+            return VibrationalOp(("I_0*0", 0), self._num_modes, self._num_modals)
+        return VibrationalOp(list(zip(label_list[non_zero].tolist(), coeff_list[non_zero])),
+                             self._num_modes, self._num_modals)
+
+    def compose(self, other: "VibrationalOp") -> "VibrationalOp":
+        if isinstance(other, VibrationalOp):
+            # Initialize new operator_list for the returned Vibration operator
+            new_data = []
+
+            # Compute the product (Vibration type operators consist of a sum of VibrationalOp):
+            # F1 * F2 = (B1 + B2 + ...) * (C1 + C2 + ...) where Bi and Ci are VibrationalOp labels
+            for label1, cf1 in self.to_list():
+                for label2, cf2 in other.to_list():
+                    new_label, not_empty = self._single_mul(label1, label2)
+                    if not not_empty:
+                        # empty operator
+                        continue
+                    new_data.append((new_label, cf1 * cf2))
+
+            if not new_data:
+                return VibrationalOp(("I_0*0", 0), self._num_modes, self._num_modals)
+
+            return VibrationalOp(new_data, self._num_modes, self._num_modals)
+
+        raise TypeError(
+            f"Unsupported operand type(s) for *: 'VibrationalOp' and '{type(other).__name__}'"
+        )
+
+    # Map the products of two operators on a single vibrational mode to their result.
+    _MAPPING = {
+        # 0                   - if product vanishes,
+        # new label           - if product does not vanish
+        ("I", "I"): "I",
+        ("I", "+"): "+",
+        ("I", "-"): "-",
+        ("I", "N"): "N",
+        ("I", "E"): "E",
+        ("+", "I"): "+",
+        ("+", "+"): "0",
+        ("+", "-"): "N",
+        ("+", "N"): "0",
+        ("+", "E"): "+",
+        ("-", "I"): "-",
+        ("-", "+"): "E",
+        ("-", "-"): "0",
+        ("-", "N"): "-",
+        ("-", "E"): "0",
+        ("N", "I"): "N",
+        ("N", "+"): "+",
+        ("N", "-"): "0",
+        ("N", "N"): "N",
+        ("N", "E"): "0",
+        ("E", "I"): "E",
+        ("E", "+"): "0",
+        ("E", "-"): "-",
+        ("E", "N"): "0",
+        ("E", "E"): "E",
+    }
+
+    @classmethod
+    def _single_mul(cls, label1: str, label2: str) -> Tuple[str, bool]:
+        if len(label1) != len(label2):
+            raise QiskitNatureError("Operators act on Fermion Registers of different length")
+
+        new_label = []
+
+        for pair in zip(label1, label2):
+            # Check what happens to the symbol
```

Comment:
What does this mean? If its stating what the code is doing or should be doing can it not be more explicit? Like check if symbol becomes 0 which is discarded. Check what happens is just so abstract

Reply:
These comments were copied over from `FermionicOp`. I will update them here and in that class, too.